### PR TITLE
Geolocation map

### PR DESCRIPTION
**Pull request by ichha132**

### CODE REVIEW

File: frontend/src/views/merchantsList.vue [hunk 1]
```diff
@@ -1,59 +1,168 @@
...
+    <gmap-map :center="center" :zoom="17" style="width:100%;  height: 400px;">
+      <gmap-marker
+        :key="index"
+        v-for="(m, index) in markers"
+        :position="m.position"
+        icon="http://maps.google.com/mapfiles/ms/icons/green-dot.png"
+      />
+
+      <gmap-marker
+        :key="index2"
```
Comment (pulkit-singhal) [inline frontend/src/views/merchantsList.vue:46]: please use **index** as the key

File: frontend/src/views/merchantsList.vue [hunk 2]
```diff
@@ -1,59 +1,168 @@
...
       </div>
     </div>
-    <div id="delivery" class="card">
-      <p>Click on merchant to get more details and placing the order</p>
-    </div>
-  </div>
+    <gmap-map :center="center" :zoom="17" style="width:100%;  height: 400px;">
+      <gmap-marker
+        :key="index"
+        v-for="(m, index) in markers"
```
Comment (pulkit-singhal) [inline frontend/src/views/merchantsList.vue:40]: Use **marker** instead of **m**

File: frontend/src/views/merchantsList.vue [hunk 3]
```diff
@@ -1,59 +1,168 @@
...
+      />
+    </gmap-map>
   </div>
 </template>
 
 <script>
 import firebase from "firebase";
 import Logout from "@/components/Logout.vue";
+import * as Geofire from "geofire";
+var j;
```
Comment (pulkit-singhal) [inline frontend/src/views/merchantsList.vue:59]: Why this **var j**

File: frontend/src/views/merchantsList.vue [hunk 4]
```diff
@@ -1,59 +1,168 @@
...
+        [19.0279532, 72.88076479],
+        [19.02887438, 72.88079917],
+        [19.02777, 72.8804372],
+        [19.0299372, 72.880763762],
+        [19.0286372, 72.88057482],
+        [27.2038, 77.5011],
+        [19.12399 , 72.9092],
+        [19.1234, 72.914271],
+        [19.1235,72.9092]
+      ],
```
Comment (pulkit-singhal) [inline frontend/src/views/merchantsList.vue:100]: This should not be put here!

File: frontend/src/views/merchantsList.vue [hunk 5]
```diff
@@ -1,59 +1,168 @@
...
     Logout
   },
   data() {
     return {
-      merchants: [],timeString:""
+      merchants: [],
+      timeString: "",
+      center: { lat: 45.508, lng: -73.587 },
+      markers: [],
+      markers2: [],
```
Comment (pulkit-singhal) [inline frontend/src/views/merchantsList.vue:76]: Can you please change marker2 to something meaningful like, merchantLocationMarkers

File: frontend/src/views/merchantsList.vue [hunk 6]
```diff
@@ -1,59 +1,168 @@
...
-  components:{
+  components: {
     Logout
   },
   data() {
     return {
-      merchants: [],timeString:""
+      merchants: [],
+      timeString: "",
+      center: { lat: 45.508, lng: -73.587 },
```
Comment (pulkit-singhal) [inline frontend/src/views/merchantsList.vue:74]: center should come from the GeoLocation API

File: frontend/src/views/merchantsList.vue [hunk 7]
```diff
@@ -1,59 +1,168 @@
...
+  mounted() {
+    this.FirebaseRef = firebase
+      .database()
+      .ref()
+      .push();
+    this.geoFireRef = new Geofire.GeoFire(this.FirebaseRef);
+
+    for (j = 0; j < this.MerchantLocations.length; j++) {
+      this.geoFireRef.set("Merchant" + j, this.MerchantLocations[j]);
+    }
```
Comment (pulkit-singhal) [inline frontend/src/views/merchantsList.vue:115]: This should be done before!

File: frontend/src/views/merchantsList.vue [hunk 8]
```diff
@@ -1,59 +1,168 @@
...
   methods: {
     ItemDetails(merchantdetails) {
       merchantexp = merchantdetails;
       itemexp = merchantdetails.itemDetails;
-      time=this.timeString;
+      time = this.timeString;
       this.$router.push("/orderDetails");
+    },
+    addMarker() {
+      var i;
```
Comment (pulkit-singhal) [inline frontend/src/views/merchantsList.vue:128]: define var inside the loop

File: frontend/src/main.js [hunk 9]
```diff
@@ -5,7 +5,8 @@ import './registerServiceWorker'
 import router from './router'
 import VueMeta from 'vue-meta';
 import moment from 'moment';
-
+import * as VueGoogleMaps from "vue2-google-maps";
+import {config} from '../googleMap-config.js';
```
[the remark this answers is not in the document]
Comment (ichha132) [reply frontend/src/main.js:9]: I am sorry, but I didn't understand the problem here

[the remark this answers is not in the document]
Comment (pulkit-singhal) [reply frontend/src/main.js:9]: As a good JS practice, there is a spacing in the parentheses for more clarity!

File: frontend/src/views/itemdetails.vue [hunk 10]
```diff
@@ -69,6 +69,12 @@
           >Confirm Order</button>
         
         </div>
+=======
```
Comment (pulkit-singhal) [inline frontend/src/views/itemdetails.vue:72]: This is not expected!

**Pull request merged**